plugins/scaffolder-field-extensions/scaffolder-frontend-module-http-request-field/src/components/SelectFieldFromApi/SelectFieldFromApi.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import { FieldProps, FieldValidation } from '@rjsf/core';
+import FormControl from '@material-ui/core/FormControl';
+import { KubernetesValidatorFunctions } from '@backstage/catalog-model';
+import {
+  discoveryApiRef,
+  fetchApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import {
+  ErrorPanel,
+  Progress,
+  Select,
+  SelectItem,
+} from '@backstage/core-components';
+import { useAsync } from 'react-use';
+import { get } from 'lodash';
+
+export const SelectFieldFromApi = (props: FieldProps<string>) => {
+  const discoveryApi = useApi(discoveryApiRef);
+  const fetchApi = useApi(fetchApiRef);
+  const [dropDownData, setDropDownData] = useState<SelectItem[] | undefined>();
+
+  const { error } = useAsync(async () => {
+    const baseUrl = await discoveryApi.getBaseUrl('');
+    const params = new URLSearchParams(props.uiSchema['ui:params']);
+    const response = await fetchApi.fetch(
+      `${baseUrl}${props.uiSchema['ui:path']}?${params}`,
+    );
+    const body = await response.json();
+    const array = get(body, props.uiSchema['ui:arraySelector']);
+    setDropDownData(
+      array.map((item: any) => {
+        const value = get(item, props.uiSchema['ui:valueSelector']);
+        const label = get(item, props.uiSchema['ui:labelSelector']);
+        return {
+          value,
+          label: label || value,
+        };
+      }),
+    );
+  });
+
+  if (error) {
+    return <ErrorPanel error={error} />;
+  }
+
+  if (!dropDownData) {
+    return <Progress />;
+  }
+
+  return (
+    <FormControl
+      margin="normal"
+      required={props.required}
+      error={props.rawErrors?.length > 0 && !props.formData}
+    >
+      <Select items={dropDownData} label="Select" onChange={props.onChange} />
+    </FormControl>
+  );
+};
+
+export const selectFieldFromApiValidation = (
```

Comment:
I should probably delete this function. I dont think it serves a purpose.?

Reply:
agree

Reply:
Actually turns out its not useless. It is validating the that field has responded with valid data before returning it.